tests/modules/programs/borgmatic/patterns-configuration.nix

```diff
@@ -0,0 +1,147 @@
+{ config, pkgs, ... }:
+
+let
+
+  boolToString = bool: if bool then "true" else "false";
+  backups = config.programs.borgmatic.backups;
+
+in {
+  programs.borgmatic = {
+    enable = true;
+    backups = {
+      main = {
+        location = {
+          patterns = [
+            "R my-stuff-to-backup"
+            "- *.swp"
+            "+ my-stuff-to-backup/junk/important_file"
+            "- my-stuff-to-backup/junk"
+          ];
+          repositories = [
+            "/mnt/disk1"
```

Comment:
Most of these configuration options are already tested by basic-configuration.nix. To reduce maintenance effort, I would only test location.patterns here and remove all other options and checks.